### ISSUE TITLE
Update endogenous variables without re-fitting model

### ISSUE THREAD

**Issue by lgnashold** [alkaline-ml/pmdarima#344]
**Question**
I have split a time series into a training set and a test set. I fit my model on my training set. 

I am looking for a way to test my data one point at a time, using every previous point in the model (essentially, I want to do cross validation on my test data). However, I _do not_ want to refit my model for every test point. I simply want to fit my model once, but be able to add endogenous data later (my motivation is that fitting many times is prohibitively slow since my m is relatively high). 

So far, I have tried using the CV module and the update method for the ARIMA model. Even with maxiter = 0, however, the update method still refits the data. The CV module also refits the data at every point. I also tried using the out_of_sample_size parameter, but that did not seem to be what I was looking for either.

The documentation around the update method hints that this is possible. I was wondering if I missed something, or if this is just not supported?

**Comment by tgsmith61591** [comment]
Hey, this is a great question and one that I wish I had a better answer for. When I set out to write the `update` method, my goal was to take the parameters learned on the train method, and simply "update" the model with the test observations so that forecasts started from the proper point. What I discovered in trying to do this is that there is some magic in the statsmodels internals I couldn't get around by simply grafting the parameters into a new model instance. The `maxiter` approach I've recommended is how I sped up the process, and it has the added benefit that it won't completely refit the entire model; it will only take `maxiter` new steps **starting from the trained model parameters**. This is the closest I could get to solving the same problem you're inquiring about.

It's possible that later updates to the statsmodels API has made this possible, and it's certainly something worth looking back into as we start planning our next release.
